inst/@sym/solve.m

```diff
@@ -116,35 +137,22 @@
             '    d = d[0]'
             'return d,' };
 
-    out = python_cmd (cmd, varargin{:});
+    out = python_cmd ([strrep(base, '{extra}', ', dict=True'); cmd], varargin{:});
 
     varargout = {out};
 
   else  % multiple outputs
-    cmd = { 'eqs = list(); symbols = list()'
-            'for arg in _ins:'
-            '    if arg.is_Relational:'
-            '        eqs.append(arg)'
-            '    else:'
-            '        symbols.append(arg)'
-            '#'
-            'if len(symbols) > 0:'
-            '    (vars, solns) = sp.solve(eqs, symbols, set=True)'
-            'else:'
-            '    (vars, solns) = sp.solve(eqs, set=True)'
-            '#'
-            'd = []'
+    cmd = { '(vars, solns) = d'
+            'q = []'
             'for (i, var) in enumerate(vars):'
-            '    d.append(sp.Matrix([t[i] for t in solns]))'
-            'return d,' };
+            '    q.append(sp.Matrix([t[i] for t in solns]))'
+            'return q,' };
 
-    out = python_cmd (cmd, varargin{:});
+    out = python_cmd ([strrep(base, '{extra}', ', set=True'); cmd], varargin{:});
 
-    varargout = out;
+    assert( isequal( nargout, length(out)), ['This system have ' num2str(length(out)) ' solution, favor set correct numbers of outputs.']);
 
-    if (length(out) ~= nargout)
-      warning('solve: number of outputs did not match solution vars');
-    end
```

Comment:
I also don't understand this change: @latot explains above but I don't understand if its a bug fix, strengthening warning to error, or just adding some more information.

Also `s/favor/please/`